src/grid/onedgrid.py

```diff
@@ -165,221 +177,165 @@ def GaussChebyshevType2(npoints):
     OneDGrid
-        A 1D grid instance.
+        A 1-D grid instance containing points and weights.
 
     """
     if npoints <= 1:
-        raise ValueError("npoints must be greater that one, given {npoints}")
-    idx = np.arange(npoints) + 1
-    points = idx / (npoints + 1)
-
-    weights = np.zeros(npoints)
+        raise ValueError(f"Argument npoints must be an integer > 1, given {npoints}")
 
-    m = np.arange(npoints) + 1
+    points = np.arange(1, npoints + 1, 1) / (npoints + 1)
 
-    bm = (np.ones(npoints) - np.cos(m * np.pi)) / (m * np.pi)
+    # make 1-D array of m values going from 1 to n
+    m = np.arange(1, npoints + 1, 1)
+    bm = (1.0 - np.cos(m * np.pi)) / (m * np.pi)
+    # make 2-D array of sin(pi * m * xi) where rows/columns correspond to different m/xi
     sim = np.sin(np.outer(m * np.pi, points))
+    # multiply 2 matrices (bm is treated like a (1, n) matrix)
     weights = bm @ sim
+    weights *= 2 / (npoints + 1)
 
+    # change integration range using variable q = 2x - 1
     points = 2 * points - 1
-    weights *= 4 / (npoints + 1)
+    weights *= 2
 
     return OneDGrid(points, weights, (-1, 1))
 
 
 def RectangleRuleSine(npoints):
-    r"""Generate 1D grid on (0:1) interval based on rectangle rule.
-
-    The fundamental definition of this quadrature is:
-
-    .. math::
-        \int_{0}^{1} f(x) dx \approx \sum_{i=1}^n w_i f(x_i)
-
-    The range of integration can be modified by :math: `q = 2 x - 1`.
-
-    .. math::
-        2 \int_{0}^{1} f(x) dx = \int_{-1}^{1} f(q) dq
-
-    Where
+    r"""Generate 1-D grid on [-1, 1] interval using Interior Rectangle Rule for Sines.
 
     .. math::
-        x_i = \frac{2 i - 1}{2 n}
+       \int_{0}^{1} f(x) dx \approx& \sum_{i=1}^n w_i f(x_i) \\
+       x_i =& \frac{2 i - 1}{2 n} \\
+       w_i =& \frac{2}{n^2 \pi} \sin(n\pi x_i) \sin^2(n\pi /2) +
+              \frac{4}{n \pi} \sum_{m=1}^{n-1} \frac{\sin(m \pi x_i)\sin^2(m\pi /2)}{m}
 
-    And the weights
+    For consistency with other 1-D grids, the integration range is modified by :math:`q=2x-1`, and
 
     .. math::
-        w_i = \frac{2}{n^2 \pi} \sin(n\pi x_i) \sin^2(n\pi /2)
-            + \frac{4}{n \pi}\sum_{m=1}^{n-1}
-            \frac{\sin(m \pi x_i)\sin^2(m\pi /2)}
-                {m}
+       2 \int_{0}^{1} f(x) dx = \int_{-1}^{1} f(q) dq
 
     Parameters
     ----------
     npoints : int
-        Number of points in the grid.
+        Number of grid points.
 
     Returns
     -------
     OneDGrid
-        A 1D grid instance.
+        A 1-D grid instance containing points and weights.
 
     """
     if npoints <= 1:
-        raise ValueError("npoints must be greater that one, given {npoints}")
-    idx = np.arange(npoints) + 1
-    points = (2 * idx - 1) / (2 * npoints)
+        raise ValueError(f"Argument npoints must be an integer > 1, given {npoints}")
+
+    points = (2 * np.arange(1, npoints + 1, 1) - 1) / (2 * npoints)
 
     weights = (
         (2 / (npoints * np.pi ** 2))
```

Comment:
@rayhe88 & @tczorro: Checking Eq. (6) in **1Dgrids.pdf** (which is also included in the docstring), I think the denominator should be `npoints**2 * np.pi`, is this a bug, or the equations were wrong? If the latter, can you share the reference?

Reply:
@FarnazH  I'm afraid it is a bug, I checked the equations and the reference again. And yes it is `np.pi * npoints ** 2 `

Reply:
Opened #120 to fix this bug.